### PR TITLE
A locked profile is sufficient to skip the first-run dialog.

### LINKED ISSUE #4252: Deployment profiles: if there's locked data but no defaults data, there should be no first-run dialog
Details in heading

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #4252

### CODE REVIEW

**Review by mook-as — DISMISSED** [review]
The comment block could be improved, because it doesn't ever actually _say_ what the effect of having a non-empty deployment profile is (and it hasn't been updated to mention locked profiles).

File: pkg/rancher-desktop/config/settings.ts [hunk 1]
```diff
@@ -230,9 +230,11 @@ export function load(deploymentProfiles: DeploymentProfileType): Settings {
...
-        _.merge(settings, deploymentProfiles.defaults);
-        if (!_.has(deploymentProfiles.defaults, 'virtualMachine.memoryInGB')) {
+      if (Object.keys(deploymentProfiles.defaults).length || Object.keys(deploymentProfiles.locked).length) {
+        if (Object.keys(deploymentProfiles.defaults).length) {
+          _.merge(settings, deploymentProfiles.defaults);
+        }
+        if (!_.has(settings, 'virtualMachine.memoryInGB') && !_.has(deploymentProfiles.locked, 'virtualMachine.memoryInGB')) {
           setDefaultMemory = true;
         }
       } else {
```
Comment (jandubois) [inline pkg/rancher-desktop/config/settings.ts:240]: I don't understand why the change is needed, and I don't understand why you need to check for `keys` at all: isn't the merge a no-op when you merge in an empty object?
And how does having a locked profile affect the merging at all?

Wouldn't do this perform the same action:

```suggestion
      _.merge(settings, deploymentProfiles.defaults);
      if (!_.has(settings, 'virtualMachine.memoryInGB') && !_.has(deploymentProfiles.locked, 'virtualMachine.memoryInGB')) {
        setDefaultMemory = true;
      } else {
```

Comment (ericpromislow) [reply pkg/rancher-desktop/config/settings.ts:240]: Yes, that's correct. I thought of this biking home, and arrived at the same code.

**Review by mook-as — APPROVED** [review]

File: pkg/rancher-desktop/config/settings.ts [hunk 2]
```diff
@@ -226,13 +226,18 @@ export function load(deploymentProfiles: DeploymentProfileType): Settings {
   } catch (err: any) {
     settings = clone(defaultSettings);
     if (err.code === 'ENOENT') {
```
Comment (jandubois) [inline pkg/rancher-desktop/config/settings.ts:228]: I think it is all correct now, but I still find the logic quite complex. Wouldn't this be equivalent and much clearer (pseudo-code):

```
     if (err.code === 'ENOENT') {
       _.merge(settings, deploymentProfiles.defaults);
       if (!Object.keys(deploymentProfiles.defaults).length && !Object.keys(deploymentProfiles.locked).length) {
         _isFirstRun = true;
       }
     }
     _.merge(settings, deploymentProfiles.locked);
     if (!_.has(settings, 'virtualMachine.memoryInGB') {
       # ... set default memory
     }
     # disable application updater for dev versions
     save(settings);
```

Comment (jandubois) [reply pkg/rancher-desktop/config/settings.ts:228]: I'm going to merge the change as-is (as it seems to be correct), but please simplify it in a separate PR!

**Review by jandubois — APPROVED** [review]
Thanks!
